sandstone/filewatcher.py

```diff
@@ -0,0 +1,43 @@
+from watchdog.observers import Observer
+import watchdog
+
+class Filewatcher(watchdog.events.FileSystemEventHandler):
+    """
+    Starts a watchdog instance to watch over the filesystem
+    Handles events - CREATE, REMOVE,MOVE
+    """
+    watched_directories = []
+
+    def __init__(self):
+        print 'Watchdog instance started'
+
+    def add_directory_to_watch(self, directory):
+        if directory not in self.watched_directories:
+            self.watched_directories.append(directory)
+        # Add the watcher
+        observer = Observer()
+        observer.schedule(self, directory, recursive=False)
+        observer.start()
+
+    def remove_directory_to_watch(self, directory):
+        if directory in self.watched_directories:
+            self.watched_directories.remove(directory)
+        # TODO remove the watchers if needed
```

Comment:
+1 on this TODO. A list of running observers should be maintained as a static member of this class, so that the observer can be stopped when its directory is removed.